Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -176,6 +182,27 @@ public async Task<bool> UserHasProblemPermissions(int problemId, string? userId,
             return await this.contestsBusiness.UserHasContestPermissions(problem.ContestId, userId, isUserAdmin);
         }
 
+        public async Task ReevaluateProblemsByOrderBy(int contestId, Problem problem)
+        {
+            var problemsGroups = this.problemGroupData.GetAllVisibleByContest(contestId);
+
+            if (problemsGroups.All(pg => pg.Problems.Count(p => !p.IsDeleted) == 1))
+            {
+               await this.problemGroupsBusiness.ReevaluateProblemsAndProblemGroupsByOrderBy(contestId, problem.ProblemGroup);
+            }
+            else
+            {
+                // We detach the existing entity, in order to avoid tracking exception on Update.
+                this.problemGroupData.Detach(problem.ProblemGroup);
+
+                var problems = problemsGroups.FirstOrDefault(pg => pg.Problems
+                                                    .Any(p => p.Id == problem.Id))?.Problems
+                                                    .Where(p => !p.IsDeleted);
+
+                await this.problemsOrderableService.ReevaluateOrderBy(problems!);
```

Comment:
I like `ReevaluateOrder` better, as with `By` at the end seems like something is missing